src/collaboration-platform/GitHubCollaborationPlatform.ts

```diff
@@ -17,25 +41,126 @@ export class GitHubCollaborationPlatform implements CollaborationPlatform {
   }
 
   async getPRForCommit(commit: Commit): Promise<PullRequestInfo | null> {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const {
+        data,
+      } = await this.octokit.repos.listPullRequestsAssociatedWithCommit({
+        owner,
+        repo,
+        commit_sha: commit.hash,
+      });
+      const prResult = data[0];
+      return {
+        number: prResult.number,
+        url: prResult.url,
+        title: prResult.title,
+        description: prResult.body,
+        isOutdated: prResult.head.sha !== commit.hash,
+      };
+    } catch (e) {
+      return null;
+    }
   }
 
   async getPR(prNumber: PullRequestID): Promise<PullRequestInfo | null> {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const prResult = this.octokit.pulls.get({
+        owner,
+        repo,
+        pull_number: prNumber,
+      });
+
+      return {
+        number: prNumber,
+        url: (await prResult).data.url,
+        title: (await prResult).data.title,
+        description: (await prResult).data.body,
+      };
+    } catch (e) {
```

Comment:
Are we expecting anything to be thrown? I can't tell from Octokit's online docs. If we're expecting the promise to resolve but with an error, we'll want to make a check for the truthiness of `pullRequest` above.

I'm just trying to avoid unnecessarily using such big try/catch blocks because it's often much easier to debug a crash than a silent failure.